pkg/virt-controller/watch/vm/vm.go

```diff
@@ -863,17 +870,29 @@ func (c *Controller) handleVolumeUpdateRequest(vm *virtv1.VirtualMachine, vmi *v
 	case *vm.Spec.UpdateVolumesStrategy == virtv1.UpdateVolumesStrategyMigration:
 		// Validate if the update volumes can be migrated
 		if err := volumemig.ValidateVolumes(vmi, vm, c.dataVolumeStore, c.pvcStore); err != nil {
-			log.Log.Object(vm).Errorf("cannot migrate the VM. Volumes are invalid: %v", err)
+			if errors.Is(err, storagetypes.ErrPVCNotFound) || errors.Is(err, storagetypes.ErrDVNotFound) {
+				msg := fmt.Sprintf("One of the destination volumes doesn't exist: %v", err)
+				log.Log.Object(vm).Error(msg)
+				if err := volumemig.DisableVolumeMigration(c.clientset, vmi, msg); err != nil {
+					return err
+				}
+				return nil
+			}
 			setRestartRequired(vm, err.Error())
+			log.Log.Object(vm).Errorf("cannot migrate the VM. Volumes are invalid: %v", err)
 			return nil
+		} else if err := volumemig.RemoveVolumeMigrationCondition(c.clientset, vmi); err != nil {
+			return err
 		}
+
 		migVols, err := volumemig.GenerateMigratedVolumes(c.pvcStore, vmi, vm)
 		if err != nil {
 			log.Log.Object(vm).Errorf("failed to generate the migrating volumes for vm: %v", err)
 			return err
 		}
 		if err := volumemig.ValidateVolumesUpdateMigration(vmi, vm, migVols); err != nil {
-			log.Log.Object(vm).Errorf("cannot migrate the VMI: %v", err)
+			msg := fmt.Sprintf("cannot migrated the VMI: %v", err)
```

Comment:
I dont see this msg being used other for the log is there a reason to split it to two lines?
Also type cannot **migrated** the VMI

Reply:
Done